seisbench/models/base.py

```diff
@@ -1519,273 +1478,363 @@ def _cut_fragments_point(self, t0, block, stations, argdict):
-            metadata = (t0, s, len(starts), stations, bucket_id)
-            yield window, metadata
 
-    async def _async_reassemble_blocks_point(self, queue_in, queue_out, argdict):
+        return [
+            TraceSegment(
+                data=group.data[..., offset : offset + self.in_samples],
+                key=(group.start_time.timestamp, "__".join(group.stations)),
+                start_time=group.start_time,
+                window_offset=int(offset),
+                n_windows=len(offsets),
+                stations=group.stations,
+                bucket_id=bucket_id,
+                in_samples=in_samples,
+                pred_sample=pred_sample,
+            )
+            for offset in offsets
+        ]
+
+    async def _iter_reassemble_predictions_point(
+        self,
+        prediction_segments: AsyncGenerator[list[PredictionSegment]],
+        argdict,
+    ) -> AsyncGenerator[PredictionsStacked]:
         """
         Wrapper with queue IO functionality around :py:func:`_reassemble_blocks_point`
         """
-        buffer = defaultdict(list)  # Buffers predictions until a block is complete
 
-        elem = await queue_in.get()
-        while elem is not None:
-            output = self._reassemble_blocks_point(elem, buffer, argdict)
-            if output is not None:
-                await queue_out.put(output)
+        out_queue = asyncio.Queue()
+
+        async def worker():
+            async for segment in prediction_segments:
+                out = self._reassemble_blocks_point(segment, argdict)
+                await out_queue.put(out)
+                await asyncio.sleep(0)  # Yield control to event loop
 
-            elem = await queue_in.get()
+            await out_queue.put(None)
 
-        await queue_out.put(None)
+        task = asyncio.create_task(worker())
+        while True:
+            ret = await out_queue.get()
+            if ret is None:
+                break
+            yield ret
+            out_queue.task_done()
+        await task
 
-    def _reassemble_blocks_point(self, elem, buffer, argdict):
+    def _reassemble_blocks_point(
+        self,
+        segments: list[PredictionSegment],
+        argdict: dict,
+    ) -> PredictionsStacked:
         """
         Reassembles point predictions into numpy arrays. Returns None except if a buffer was processed.
         """
         stride = self._argdict_get_with_default(argdict, "stride")
 
-        window, metadata = elem
-        t0, s, len_starts, stations, bucket_id = metadata
-        key = f"{t0}_{'__'.join(stations)}"
+        # Sort by start (overwrite keys to make sure window is never used as key)
+        segments = sorted(segments, key=lambda seg: seg.window_offset)
+        meta = segments[0]
+        data = np.array([seg.data for seg in segments])
+        if data.ndim == 1:
+            data = data.reshape(-1, 1)
+        pred_time = meta.start_time + self.pred_sample / argdict["sampling_rate"]
```

Comment:
This won't fail for any currently implemented model, but in principle, this should be `meta.pred_sample`. At this point, you'll get an error though from the changes in `_get_in_pred_samples` (see below).

Reply:
changed. please check